[FX-800] Expose the deferPaymentCapture method (#139)
teamforage/forage-android-sdk: README.md

- [Tokenizing an EBT Card](#tokenizing-an-ebt-card)
- [Performing a balance check](#performing-a-balance-check)
- [Capturing a payment](#capturing-a-payment)

> - [Collecting the pin for a payment](#collecting-the-pin-for-a-payment)

Reviewer: ```suggestion
- [Collecting the pin for a deferred payment](#collecting-the-pin-for-a-deferred-payment)
```

> - [Submitting the pin for a deferred payment](#submitting-the-pin-for-a-deferred-payment)

Reviewer 2: ```suggestion
- [Submitting the PIN for a deferred payment](#submitting-the-pin-for-a-deferred-payment)
```

1. [Tokenizing an EBT Card](#tokenizing-an-ebt-card)
2. [Performing a balance check](#performing-a-balance-check)
3. [Capturing a payment](#capturing-a-payment)

> 4. [Collecting the pin for a payment](#collecting-the-pin-for-a-payment)

Reviewer: ```suggestion
4. [Collecting the pin for a deferred payment](#collecting-the-pin-for-a-deferred-payment)
```

> 4. [Submitting the pin for a deferred payment](#submitting-the-pin-for-a-deferred-payment)

Reviewer 2: ```suggestion
4. [Submitting the PIN for a deferred payment](#submitting-the-pin-for-a-deferred-payment)
```

        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {

>         // establish bindings to ForagePINEditText

Reviewer: nice catch!

It is the `funding_type` on a [Payment object](#step-0-create-a-payment-object)
where you indicate whether a `Payment` will capture (SNAP or EBT Cash).

> ## Collecting the pin for a payment

Reviewer 2: ```suggestion
## Collecting the PIN for a deferred EBT payment
```

### Step 0: Create a `Payment` object

> Send a POST request to Forage's `/payments/` endpoint [to create a

Reviewer 2: ```suggestion
Send a `POST` request to Forage's `/payments/` endpoint [to create a
```

`Payment` object](https://docs.joinforage.app/reference/create-a-payment).

> You need the `ref` of the `Payment` within the POST's response for Step 3.

Reviewer 2: ```suggestion
You need the `ref` of the `Payment` within the `/payments/` response for Step 3.
```

### Step 1: Add the `ForagePINEditText` UI component

As with [performing a balance check](#performing-a-balance-check), you will need
to add a `ForagePINEditText` to your UI. Please reference
[Step 1 of Performing a Balance Check](#step-1-add-the-foragepinedittext-ui-component)

### Step 2: Customizing `ForagePINEditText`

Please reference [Step 2 of Performing a Balance Check](#step-2-customizing-foragepinedittext)

### Step 3: Call `setForageConfig()`

Please reference [Step 3 of Performing a Balance Check](#step-3-call-setforageconfig)

### Step 4: Collect the pin for the EBT payment

> The ForageSDK exposes the following function to collect a pin for an EBT payment and defer capture to the server:

Reviewer 2: ```suggestion
The ForageSDK provides a method to collect a customer's PIN for an EBT payment and defer the capture of the payment to the server.
```

> The ForageSDK provides a method to collect a customer's PIN for an EBT payment and defer the capture of the payment to the server.

Reviewer: This languages is great 💯

```kotlin

> data class CollectPinParams(

Reviewer: Thoughts on naming the params `DeferredCaptureParams`? Can do another name but `CollectPinParams` still throws me off a bit. Lmk what you think though
```suggestion
data class DeferredPaymentCaptureParams(
```

- `CollectPinParams.paymentRef`: The `paymentRef` parameter is a string identifier that refers to an instance in Forage's database of a [`Payment`](https://docs.joinforage.app/reference/create-a-payment)

#### Example

```kotlin
@AndroidEntryPoint
class PinCollectFragment : Fragment() {
    private val viewModel: PinCollectViewModel by viewModels()
    private var _binding: PinCollectFragmentBinding? = null
    private val binding get() = _binding!!

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        // establish bindings to ForagePANEditText
        _binding = PinCollectFragmentBinding.inflate(inflater, container, false)
        val snapEditText = binding.snapPinEditText
        val cashEditText = binding.cashPinEditText

        // immediately call setForageConfig() on the binding
        val forageConfig = ForageConfig(
            merchantId = viewModel.merchantAccount,
            sessionToken = viewModel.bearer
        )
        snapEditText.setForageConfig(forageConfig)
        cashEditText.setForageConfig(forageConfig)

>         // then freely call other methods on ForagePANEditText binding

Reviewer 2: ```suggestion
        // then freely call other methods on ForagePINEditText binding
```
